### PR TITLE
Log error and exit if swaylock is suid with PAM backend.

### LINKED ISSUE #15: Abort if we have the suid bit but don't need it
If swaylock has been compiled with PAM and has suid, we should probably error out.

### PR DESCRIPTION
**Pull request by c-edw**
Closes #15.

I looked at how this was done on Sway. I think I know how it works - by comparing the user's ID and the effective user's ID (which might be root) we can tell if the executable has suid set.

### CODE REVIEW

File: pam.c [hunk 1]
```diff
@@ -12,6 +12,11 @@
 static char *pw_buf = NULL;
 
 void initialize_pw_backend(void) {
+	if (getuid() != geteuid() || getgid() != getegid()) {
+		swaylock_log(LOG_ERROR,
+			"swaylock has suid but doesn't require it with PAM backend");
```
Comment (ddevault) [inline pam.c]: I think this would be clearer: "swaylock is setuid, but was compiled with the PAM backend. chmod a-x $argv[0] to fix. Aborting.".

Comment (c-edw) [reply pam.c]: Should I pass `argv` into `initialize_pw_backend` or is there another way?

Comment (ddevault) [reply pam.c]: Yeah, I would.

Comment (c-edw) [reply pam.c]: Wait, did you mean `a-s`?

Comment (ddevault) [reply pam.c]: Yeah, derp

**Comment by ddevault** [comment]
You can see what the CI does here:

https://builds.sr.ht/api/jobs/22545/manifest

It's probably that it builds both with and without PAM support.

**Comment by c-edw** [comment]
Yep, it was failing with PAM disabled. I'll be sure to check these manifests in future. Thanks :+1:

**Comment by ddevault** [comment]
Thanks!

**Pull request merged**